magicarp/pipeline/harmless_base_pipeline.py

```diff
@@ -0,0 +1,48 @@
+import json
+import gzip
+from magicarp.pipeline import Pipeline, register_datapipeline
+
+@register_datapipeline
+class HarmlessBasePipeline(Pipeline):
+    def __init__(self, path):
+        self.path = path
+        self.data = []
+    
+    def __getitem__(self, idx):
+        with gzip.open(self.path) as f:
+            for line in f:
+                data.append(json.loads(line))
+        
+        listr = data[idx]['chosen'].split(" ")
+        listr1 = data[idx]['rejected'].split(" ")
+
+        j = 0
+        for wordi in range(0, len(listr)):
+            if 'Assistant:' in listr[wordi]:
+                j = wordi
+
+        supp = listr[j].split("\n")
+        prompt = listr[:j]
+        prompt.extend(supp[:-1])
+        chosen = listr[j+1:]
+        chosen.insert(0,supp[-1])
+
+        k = 0
+        for wordi in range(0, len(listr1)):
+            if 'Assistant:' in listr1[wordi]:
+                k = wordi
+
+        supp = listr[j].split("\n")
+        rejected = listr1[k+1:]
+        rejected.insert(0, supp[-1])
+
+        return (prompt, chosen, rejected)
+    
+    def __len__(self):
+        length = 0
+        with gzip.open(self.path) as f:
+            for file in f:
+                length = length + 1
+        return length
+
```

Comment:
need to add a create_preprocess_fn method as well. Examples exist in the summary_rm branch example pipeline and instruct_rpcd example. The summary_rm one (examples/summary_test.py) is almost identical to what you will need as the example consists of articles, chosen tldrs, rejected tldrs.